### PR TITLE
Feature/multiple selections sc4

### PR DESCRIPTION
**Pull request by PaulWestphal**

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/FeatureTreeDeleteOperation.java [hunk 1]
```diff
@@ -44,53 +45,57 @@
...
+					final AbstractFeatureModelOperation op = new DeleteFeatureOperation(featureModel, feat);
+					operations.add(op);
 				}
-				final AbstractFeatureModelOperation op = new DeleteFeatureOperation(featureModel, feat);
-				operations.add(op);
+			} else {
+				final String containedFeatures = containedFeatureList.toString();
+				final MessageDialog dialog = new MessageDialog(new Shell(), DELETE_ERROR, FEATURE_SYMBOL,
+						"The following features are contained in constraints:" + '\n' + containedFeatures.substring(1, containedFeatures.length() - 1) + '\n' + '\n'
+							+ UNABLE_TO_DELETE_THIS_FEATURES_UNTIL_ALL_RELEVANT_CONSTRAINTS_ARE_REMOVED_,
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/FeatureTreeDeleteOperation.java:94]: Check indentation

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/CollapseFeatureOperation.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	public CollapseFeatureOperation(IFeature feature, IGraphicalFeatureModel graphicalFeatureModel, String operationLabel) {
+		super(graphicalFeatureModel.getFeatureModel(), operationLabel);
+		this.graphicalFeatureModel = graphicalFeatureModel;
+		this.feature = feature;
+	}
+
+	@Override
+	protected FeatureIDEEvent operation() {
+		if (feature.getStructure().hasChildren()) {
+
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/CollapseFeatureOperation.java]: This empty line is not needed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/editparts/FeatureEditPart.java [hunk 3]
```diff
@@ -122,7 +123,7 @@ public void performRequest(Request request) {
 		if (request.getType() == RequestConstants.REQ_DIRECT_EDIT) {
 			showRenameManager();
 		} else if (request.getType() == RequestConstants.REQ_OPEN) {
-			final SetFeatureToCollapseOperation op = new SetFeatureToCollapseOperation(feature, featureModel);
+			final CollapseFeatureOperation op = new CollapseFeatureOperation(feature, featureModel, "Collapse Operation"); //TODO: PLATZHALTER
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/editparts/FeatureEditPart.java]: "TODO: Platzhalter"?
Shouldn't the operationLabel ("Collapse Operation") be read from the Stringtable?

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MandatoryAction.java [hunk 4]
```diff
@@ -23,17 +23,20 @@
...
 import org.eclipse.ui.PlatformUI;
 
+import de.ovgu.featureide.fm.core.base.IFeature;
 import de.ovgu.featureide.fm.core.base.IFeatureModel;
 import de.ovgu.featureide.fm.ui.FMUIPlugin;
 import de.ovgu.featureide.fm.ui.editors.featuremodel.operations.SetFeatureToMandatoryOperation;
 
 /**
- * Turns a feature in an And-group into a mandatory feature.
+ * Turns a features in an And-group into a mandatory feature.
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MandatoryAction.java]: That sentence does not make a whole lot of sense

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/HiddenAction.java [hunk 5]
```diff
@@ -45,21 +48,34 @@ public HiddenAction(Object viewer, IFeatureModel featureModel) {
...
 			FMUIPlugin.getDefault().logError(e);
-
 		}
 	}
 
 	@Override
 	protected void updateProperties() {
 		setEnabled(true);
-		setChecked(feature.getStructure().isHidden());
+		// A selection of features is considered hidden iff every feature is hidden.
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/HiddenAction.java]: I don't think that every one knows "if and only if" as iff

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CreateConstraintWithAction.java [hunk 6]
```diff
@@ -76,8 +77,9 @@ public void selectionChanged(SelectionChangedEvent event) {
 	 * @param featureName
 	 */
 	protected void updateConstraintActionText(String featureName) {
-		selectedFeature = featureName;
-		setText(CREATE_CONSTRAINT + (featureName.isEmpty() ? "" : " starting with \"" + featureName + "\""));
+		selectedFeature =	featureName;
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CreateConstraintWithAction.java]: There's an unnecessary tab-stop here

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java [hunk 7]
```diff
@@ -66,41 +79,150 @@ public void selectionChanged(SelectionChangedEvent event) {
...
+	}
+	
+	private String getStringLabel() {
+		if (featureArray.length == 1) {
+			if (isEveryFeatureCollapsed()) {
+				return SET_FEATURE_EXPANDED;
+			} else {
+				return SET_FEATURE_COLLAPSED;
+			}
+		} else {
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java:158]: What happens if the featureArray.length is 0?

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/AbstractAction.java [hunk 8]
```diff
@@ -32,37 +32,50 @@
...
-
 		}
-
 	}
 
 	@Override
 	protected void updateProperties() {
 		setEnabled(true);
-		setChecked(feature.getStructure().isAbstract());
+		// A selection of features is considered abstract iff every feature is abstract.
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/AbstractAction.java]: Same as below

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java [hunk 9]
```diff
@@ -657,86 +657,80 @@ private void fillContextMenu(IMenuManager menu) {
...
+
 			menu.add(new Separator());
 			menu.add(calculateDependencyAction);
-			menu.add(reverseOrderAction);
-			menu.add(legendAction);
 			menu.add(new Separator());
-		} else if (editConstraintAction.isEnabled() && !connectionSelected) {
+
+      // if a constraint is selected
+		} else if (editConstraintAction.isEnabled()	&& !connectionSelected) {
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java]: Unnecessary tab-stop before &&

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/editparts/FeatureEditPart.java (hunk 3 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/editparts/FeatureEditPart.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/CollapseFeatureOperation.java (hunk 2 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/CollapseFeatureOperation.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MandatoryAction.java (hunk 4 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MandatoryAction.java]: fair point

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/HiddenAction.java (hunk 5 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/HiddenAction.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CreateConstraintWithAction.java (hunk 6 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CreateConstraintWithAction.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java (hunk 7 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java:158]: the action won't even show up in the menu then.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/AbstractAction.java (hunk 8 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/AbstractAction.java]: ok

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java (hunk 9 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java]: fixed

**Pull request merged**